### PR TITLE
fixed: #735, #775 toast on back pressed in home activity, sorted Allergens alphabetically

### PR DESCRIPTION
**Pull request by AbsolutelySaurabh**
## Description

1. Now showing a toast saying "Please press back again to exit" on button pressed.
2. Allergens have been sorted alphabetically.

## Related issues and discussion
#735 , #775 
 
 ## Screen-shots, if any
 
screenshot_20180216-014621

screenshot_20180217-053738

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AlertUserFragment.java [hunk 1]
```diff
@@ -74,12 +77,20 @@ public void onViewCreated(final View view, @Nullable Bundle savedInstanceState)
 
         mRvAllergens = (RecyclerView) view.findViewById(R.id.alergens_recycle);
         mAllergensEnabled = mAllergenDao.queryBuilder().where(AllergenDao.Properties.Enable.eq("true")).list();
-        mAdapter = new AllergensAdapter(mAllergensEnabled, getActivity());
+        mAdapter = new AllergensAdapter(sortedAllergensList(), getActivity());
+        Log.e("mAllergensEnabled: ", mAllergensEnabled.toString());
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AlertUserFragment.java]: This should be a debug and not an error log entry

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Allergen.java [hunk 2]
```diff
@@ -19,7 +21,7 @@
 @Entity(indexes = {
         @Index(value = "name", unique = true)
 })
-public class Allergen {
+public class Allergen implements Comparable<Allergen>{
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Allergen.java]: Can you provide Unit Tests for this class and its methods?

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,14 @@
+*.iml
 .gradle
 /local.properties
-/.idea
+/.idea/workspace.xml
```
Comment (policorn) [inline .gitignore]: unnecessary as `/.idea` is already in place

File: .gitignore [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+/.idea/libraries
 .DS_Store
 /build
-*.iml
\ No newline at end of file
+/captures
+.externalNativeBuild
+*.log
+hs_err_pid2879.log
```
Comment (policorn) [inline .gitignore]: This one should be already ignored by `*.log`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/BaseActivity.java [hunk 5]
```diff
@@ -36,4 +39,23 @@ public void setContentView(@LayoutRes int layoutResID) {
...
+
+    @Override
+    public void onBackPressed() {
+        if (doubleBackToExitPressedOnce) {
+            super.onBackPressed();
+            return;
+        }
+
+        this.doubleBackToExitPressedOnce = true;
+        Toast.makeText(this, "Please press back again to exit", Toast.LENGTH_SHORT).show();
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/BaseActivity.java]: Avoid hardcoded strings. This string must also be internationalized.

**Comment by AbsolutelySaurabh** [comment]
@teolemon , @policorn please review my latest commit as I've solved the issues mentioned.

**Comment by policorn** [comment]
Hi @AbsolutelySaurabh, this PR fix for #735 works as expected. Regarding #775 I'm missing the filtering option mentioned in the issue ticket, how difficult would it be to implement it? Can you split this PR so we can merge your fix for the exit confirmation?

**Comment by AbsolutelySaurabh** [comment]
closing this PR,  as I'll made new-one for both issues separately.

**Pull request closed without merging**